settings.gradle.kts

```diff
@@ -22,7 +22,9 @@ include("subclass",
     "errorprone",
     "programmatic-test")
 
-if (!JavaVersion.current().isCompatibleWith(JavaVersion.VERSION_17) && (System.getenv("ANDROID_SDK_ROOT") != null || File(".local.properties").exists())) {
+// https://developer.android.com/studio/command-line/variables#envar
+// https://developer.android.com/studio/build#properties-files
+if (System.getenv("ANDROID_HOME") != null || File("local.properties").exists()) {
```

Comment:
Would it run on JDKs above 11?

Reply:
I haven't had any issues on mockito project with JDK 17. I would say since the CI is green, and locally it also works fine, it's safe to use it. In case something breaks -- which is very unlikely --, someone can re-add the condition.

I have an Android app with full JDK 17 CI:
 * https://github.com/TWiStErRob/net.twisterrob.sun/blob/5fbf1f873e413a8a8fa9398effbeae636e53e133/.github/workflows/CI.yml#L19-L23
 * I'll update my other app now too to see what issues could be and understand why it is not JDK 17 yet.

Reply:
:+1: running tests on different version does make sense, building on latest - may be bleeding edge unless the latest is LTS like 17, but we shall see I think.

Reply:
Jake's point is that you build once, and test on different runtimes, same as your use will download your pre-built jar and run it on different JVMs.

Reply:
This way you can easily add newer JDK tests (19), even if your build is not supporting it yet.

Reply:
:+1:

Reply:
I am wondering if it makes sense to add JDK-17 to Android build matrix to CI, wdyt?

Reply:
Looks like there was totally no reason not to go for Java 17:
https://github.com/TWiStErRob/net.twisterrob.colorfilters/pull/134

And to continue the above conversation: https://github.com/mockito/mockito/issues/2898

Reply:
> I am wondering if it makes sense to add JDK-17 to Android build matrix to CI, wdyt?

I think it's ok not to, since the current recommended version is 11:
https://developer.android.com/studio/releases/gradle-plugin#compatibility-7-4-0